ctapipe/image/muon/muon_ring_finder.py

```diff
@@ -16,73 +15,65 @@ class MuonRingFitter(Component):
         default_value='chaudhuri_kundu'
     ).tag(config=True)
 
-    tel_type = traits.CaselessStrEnum(
-        ['LST_LST_LSTCam', 'MST_MST_NectarCam', 'MST_MST_FlashCam', 'MST_SCT_SCTCam',
-         'SST_1M_DigiCam', 'SST_GCT_CHEC', 'SST_ASTRI_ASTRICam', 'SST_ASTRI_CHEC'],
-        default_value='LST_LST_LSTCam'
-    ).tag(config=True)
-
-    def __init__(self, config=None, parent=None, **kwargs):
+    def __init__(self, teldes, config=None, parent=None, **kwargs):
+        """
+        teldes: telescope description
+                in form of event.inst.subarray.tel[telid]
+        """
         super().__init__(config, parent, **kwargs)
+        self.teldes = teldes
 
     def fitFormula(self, xc, yc, r):
-        '''taubin fit formula
+        """taubin fit formula
         reference : Barcelona_Muons_TPA_final.pdf (slide 6)
-        '''
-        upper_term = sum(
+        """
+        upper_term = (
             (
                     (np.array(self.x) - xc) ** 2 +
                     (np.array(self.y) - yc) ** 2
                     - r ** 2
             ) ** 2
-        )
-        lower_term = sum(
+        ).sum()
+
+        lower_term = (
             (
                     (np.array(self.x) - xc) ** 2 +
                     (np.array(self.y) - yc) ** 2
             )
-        )
+        ).sum()
 
         return np.abs(upper_term) / np.abs(lower_term)
 
     def initialization(self):
         """ initialization for taubin_fit with the
         initial default values, errors, and limits
         """
-        names = ['LST_LST_LSTCam', 'MST_MST_NectarCam', 'MST_MST_FlashCam', 'MST_SCT_SCTCam',
-                 'SST_1M_DigiCam', 'SST_GCT_CHEC', 'SST_ASTRI_ASTRICam', 'SST_ASTRI_CHEC']
-        taubin_r_initial = [0.02, 0.03, 0.03, 0.03, 0.03, 0.02, 0.04, 0.02]
-        taubin_error = [0.003, 0.005, 0.005, 0.005, 0.006, 0.004, 0.007, 0.004]
-        taubin_limit = [(-0.0365, 0.0365), (-0.0640, 0.0640), (-0.0625, 0.0625), (-0.0646, 0.0646),
-                        (-0.0719, 0.0719), (-0.0462, 0.0462), (-0.0848, 0.0848)]
-        taubin_params = {'Name': names, 'rInitial': taubin_r_initial,
-                         'error': taubin_error, 'limit': taubin_limit}
-        taubin_dictIndex = taubin_params['Name'].index(self.tel_type)
-        radius = taubin_params['rInitial'][taubin_dictIndex]
-        error = taubin_params['error'][taubin_dictIndex]
-        limit = taubin_params['limit'][taubin_dictIndex]
+        focal_length = self.teldes.optics.equivalent_focal_length / u.m
```

Comment:
use `to_value(u.m)` to get a pure number expressed in meter from an astropy quantity

Reply:
This avoids problems, if focal length is not in m, but e.g. in mm:
```
In [2]: import astropy.units as u
In [3]: f = 5000 * u.mm

In [4]: f =  f / u.m

In [5]: 0.5 / f
Out[5]: <Quantity 0.0001 m / mm>
```